salt/modules/keystoneng.py

```diff
@@ -0,0 +1,382 @@
+import salt.utils
+
+HAS_SHADE = False
+try:
+    import shade
+    from shade.exc import OpenStackCloudException
+    HAS_SHADE = True
+except ImportError:
+    pass
+
+
+def __virtual__():
+    if HAS_SHADE:
+        return 'keystoneng'
+    return False
```

Comment:
This should return a tuple with the reason it couldn't be loaded

`(False, 'Unable to load keystoneng, please install the python shade module')`